app/src/main/java/com/duckduckgo/app/browser/defaultbrowsing/prompts/DefaultBrowserPromptsExperimentImpl.kt

```diff
@@ -0,0 +1,341 @@
+                // If feature settings weren't cached before, deserialize and cache them now.
+                val parsedSettings = defaultBrowserPromptsFeatureToggles.parseFeatureSettings()
+                featureSettings = parsedSettings
+                parsedSettings
+            } ?: run {
+                Timber.e("Failed to obtain feature settings.")
+                return
+            }
+
+            when (currentExperimentStage) {
+                NOT_ENROLLED -> ENROLLED
+
+                ENROLLED -> {
+                    if (appDaysUsedRepository.getNumberOfDaysAppUsedSinceDate(enrollmentDateGMT) >= configSettings.activeDaysUntilStage1) {
+                        STAGE_1
+                    } else {
+                        null
+                    }
+                }
+
+                STAGE_1 -> {
+                    if (appDaysUsedRepository.getNumberOfDaysAppUsedSinceDate(enrollmentDateGMT) >= configSettings.activeDaysUntilStage2) {
+                        STAGE_2
+                    } else {
+                        null
+                    }
+                }
+
+                STAGE_2 -> {
+                    if (appDaysUsedRepository.getNumberOfDaysAppUsedSinceDate(enrollmentDateGMT) >= configSettings.activeDaysUntilStop) {
+                        STOPPED
+                    } else {
+                        null
+                    }
+                }
+
+                STOPPED, CONVERTED -> null
+            }
+        }
+
+        if (newExperimentStage != null) {
+            defaultBrowserPromptsDataStore.storeExperimentStage(newExperimentStage)
+
+            val action = experimentStageEvaluatorPluginPoint.getPlugins().first { it.targetCohort == activeCohortName }.evaluate(newExperimentStage)
+            if (action.showMessageDialog) {
+                _commands.send(OpenMessageDialog)
+            }
+            defaultBrowserPromptsDataStore.storeShowOverflowMenuItemState(action.showOverflowMenuItem)
+            defaultBrowserPromptsDataStore.storeHighlightOverflowMenuIconState(action.highlightOverflowMenu)
+        }
+    }
+
+    override fun onOverflowMenuOpened() {
+        appCoroutineScope.launch {
+            defaultBrowserPromptsDataStore.storeHighlightOverflowMenuIconState(highlight = false)
+        }
+    }
+
+    override fun onOverflowMenuItemClicked() {
+        appCoroutineScope.launch {
+            launchBestSelectionWindow()
+        }
+    }
+
+    override fun onMessageDialogShown() {
+    }
+
+    override fun onMessageDialogDismissed() {
+    }
+
+    override fun onMessageDialogConfirmationButtonClicked() {
+        appCoroutineScope.launch {
+            launchBestSelectionWindow()
+        }
+    }
+
+    override fun onMessageDialogNotNowButtonClicked() {
+    }
+
+    override fun onSystemDefaultBrowserDialogShown() {
+        browserSelectionWindowFallbackDeferred = appCoroutineScope.async {
+            delay(FALLBACK_TO_DEFAULT_APPS_SCREEN_THRESHOLD_MILLIS)
```

Comment:
some explanation for this delay would be helpful.

Reply:
There's an explanation attached to the `browserSelectionWindowFallbackDeferred` field at the top of the class, let me know if you think we need more than that.